js/modules/k6/http/request.go

```diff
@@ -125,7 +127,11 @@ func (c *Client) asyncRequest(method string, url goja.Value, args ...goja.Value)
 // a reverse dependency on js/common or goja.
 func (c *Client) processResponse(resp *httpext.Response, respType httpext.ResponseType) {
 	if respType == httpext.ResponseTypeBinary && resp.Body != nil {
-		resp.Body = c.moduleInstance.vu.Runtime().NewArrayBuffer(resp.Body.([]byte))
+		b, ok := resp.Body.([]byte)
+		if !ok {
+			panic("got an unexpected type for the response body, only []byte is accepted")
+		}
+		resp.Body = c.moduleInstance.vu.Runtime().NewArrayBuffer(b)
```

Comment:
There really is no way for this to happen apart from programming error on our side.

I am not against hte change - but arguably:
1. just a nolint would've been simpler
2. this panic should mention this is a programming error.